src/edu/washington/escience/myria/Type.java

```diff
@@ -280,6 +281,46 @@ public String getName() {
       return "Long";
     }
   },
+  /**
+   * byteArray type.
+   * */
+  BLOB_TYPE() {
+    @Override
+    public boolean filter(
+        final SimplePredicate.Op op,
+        final Column<?> bytesColumn,
+        final int tupleIndex,
+        final Object operand) {
+      final ByteBuffer v = bytesColumn.getBlob(tupleIndex);
+      return compare(op, v, (ByteBuffer) operand);
+    }
+
+    @Override
+    public String toString(final Column<?> column, final int tupleIndex) {
+
+      return "" + column.getBlob(tupleIndex);
```

Comment:
This would return the string repr of the ByteBuffer object (ByteByffer@xxx). Just want to make sure that it's your expectation, since printing out the bytes is indeed unreasonable.